lib/person.rb

```diff
@@ -0,0 +1,35 @@
+class Person
+
+  attr_reader :name,
+              :interests,
+              :supplies
+
+  def initialize(person_data)
+    @name = person_data[:name]
+    @interests = person_data[:interests]
+    @supplies = Hash.new(0)
+  end
+
+  def add_supply(supply, amount)
+    @supplies[supply] += amount
+  end
+
+  ## I am still hazey on comparing hashes and how best to access the data.
+
+  # def can_build?(craft)
```

Comment:
Here we want to iterate through all the `supplied_required` for the craft and make sure the person both has the supply and greater than or equal to the needed quantity. Something like the following

// for each supply required by a craft
// if the person doesn't have the supply, return false
// if the person doesn't have enough of the supply return false

```
craft.supplies_required.each do |supply, quantity_needed|
  if !@supply[supply.to_s]
    return false
  end
  if @supply[supply.to_s] < quantity_needed
    return false
  end
end

true
```